src/Domain.UnitTests/EarningsGeneratedEventBuilder_BuildTests.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using NUnit.Framework;
+using SFA.DAS.Apprenticeships.Types;
+using SFA.DAS.Funding.ApprenticeshipEarnings.Types;
+
+namespace SFA.DAS.Funding.ApprenticeshipEarnings.Domain.UnitTests;
+
+public class EarningsGeneratedEventBuilder_BuildTests
+{
+    private EarningsGeneratedEventBuilder _sut;
+    private ApprenticeshipCreatedEvent _apprenticeshipLearnerEvent;
+    private EarningsProfile _earningsProfile;
+    private EarningsGeneratedEvent _result;
+
+    [SetUp]
+    public void SetUp()
+    {
+        _sut = new EarningsGeneratedEventBuilder();
+
+        _apprenticeshipLearnerEvent = new ApprenticeshipCreatedEvent
+        {
+            FundingType = FundingType.NonLevy,
+            ActualStartDate = new DateTime(2022, 8, 1),
+            ApprenticeshipKey = Guid.NewGuid(),
+            EmployerAccountId = 114,
+            PlannedEndDate = new DateTime(2024, 7, 31),
+            UKPRN = 116,
+            TrainingCode = "able-seafarer",
+            FundingEmployerAccountId = 118,
+            Uln = "900000118",
+            AgreedPrice = 20000
```

Comment:
be good to use AutoFixture here for more randomised inputs

Reply:
done